pkg/controller/workload/workload_hash.go

```diff
@@ -42,12 +78,20 @@ func (h *HashName) StrToNum(str string) uint32 {
 	hash.Reset()
 	hash.Write([]byte(str))
 
+	if num, exits := h.strToNum[str]; exits {
+		return num
+	}
+
 	// Using linear probing to solve hash conflicts
 	for num = hash.Sum32(); num < math.MaxUint32; num++ {
```

Comment:
not related to your code。

If num overflows, should we iterate from 0

Reply:
I think it's necessary. It should be a ring.